elastic-agent-client.proto

```diff
@@ -112,15 +115,361 @@ message ActionResponse {
     bytes result = 4;
 }
 
+// State codes for the current state.
+enum State {
+    STARTING = 0;
+    CONFIGURING = 1;
+    HEALTHY = 2;
+    DEGRADED = 3;
+    FAILED = 4;
+    STOPPING = 5;
+}
+
+// Type of unit.
+enum UnitType {
+    INPUT = 0;
+    OUTPUT = 1;
+}
+
+// A unit that is part of a collector/shipper.
+message UnitExpected {
+    // Unique ID of the unit.
+    string id = 1;
+    // Unit type.
+    UnitType type = 2;
+    // Expected state of the unit.
+    State state = 3;
+    // Index of the either current configuration or new configuration provided.
+    uint64 config_state_idx = 4;
```

Comment:
I'm a bit puzzled by the idea that the meaning of this field changes... If this type represents what is expected of a unit, then shouldn't `config_state_idx` always be the config index it should be on?
If the Agent knows the unit is already on the expected `config_state_idx` it won't populate `config`, fine. However it should not change the meaning of `config_state_idx`. It's always the the config index the unit should be on

Reply:
What it means is the expected configuration index for that unit. So in the case its the same the unit already has the idx will be the same value as the previous with the contents of config missing. In the case that it needs to update the configuration its idx++ with the configuration provided.